content/docs/for-developers/sending-email/automating-subusers.md

```diff
@@ -0,0 +1,513 @@
+
+```bash
+curl -X POST -H "Authorization: Basic XXXXXXXXXXXXXXXX" -H "Content-Type: application/json" -d '{"name": “API KEY NAME"}' 'https://api.sendgrid.com/v3/api_keys'
+```
+or, you can create the key for your Subuser:
+
+```bash
+curl -X POST -H "Authorization: Bearer PARENT_APIKEY_HERE" 
+  -H "On-Behalf-Of: SUBUSER_NAME" \
+  -H "Content-Type: application/json" -d '{"name": “API KEY NAME"}' \
+  'https://api.sendgrid.com/v3/api_keys'
+  ```
+  
+Successful API key implementation Response:
+  
+  ```bash
+  => 201 created
+{"api_key":"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","api_key_id":"rGaWG-xBTHiBuhCLxWZ0Hg","name":"new api key","scope_set_id":"00766089-9730-4f16-8191-b8bedc5f57f5"}
+```
+
+Once this has been done, the Subuser has been created. From this point on, we’ll cover implementing Click Tracking, Open Tracking, and the Event Webhook for those Subusers.
+
+### Enable/Edit Applications
+As of right now this needs to be done directly through the account in question.
+
+1. **Click Tracking**
+
+```bash
+curl -X PATCH -H "Authorization: Basic XXXXXXXXXXXXXX" -H "Content-Type: application/json" -d '{"enabled": true,"enable_text" : true}' 'https://api.sendgrid.com/v3/tracking_settings/click'
+```
+
+Response:
+
+```bash
+=> 200 OK
+{"enable_text":true,"enabled":true}
+```
+2. **Open Tracking**
+
+```bash
+curl -X PATCH -H "Authorization: Basic XXXXXXXXXXXXXX" -H "Content-Type: application/json" -d '{"enabled": true}' 'https://api.sendgrid.com/v3/tracking_settings/open'
+```
+
+Response:
+
+```bash
+=> 200 ok
+{"enabled":true}
+```
+
+3. **Webhook** - Event Notification App
+
+```bash
+curl -X PATCH -H "Authorization: Basic XXXXXXXXXXXXXXX" -H "Content-Type: application/json" -d '{"enabled": true,"url":"https://api.keen.io/3.0/projects/526a8a5f05cd660472000012/partners/sendgrid?api_key=081da3fc30ca5b11b1f8b133de7064817b57c86d3261f140dc36243eb6829a100d4c5e38db8dd289194d4e02159bbf7ebad74a1fba1fce7a421a00c7a2f7d0f8ba5ef6ea3cece6345904d79395c35cdeaed221c3a208c897e79dd95da399a1d2d1d3e66768af929aa52d621e5bd296a5","group_resubscribe": true,"delivered": true,"group_unsubscribe": true,"bounce": true,"deferred": true,"unsubscribe": true,"processed": true,"open": true,"click": true,"dropped": true}' 'https://api.sendgrid.com/v3/user/webhooks/event/settings'
+```
+
+Response:
+
+```bash
+=> 200 ok
+{
+"bounce": true,
+"click": true,
+"deferred": true,
+"delivered": true,
+"dropped": true,
+"enabled": true,
+"group_resubscribe": true,
+"group_unsubscribe": true,
+"open": true,
+"processed": true,
+"spam_report": false,
+"unsubscribe": true,
+"url": "https://api.keen.io/3.0/projects/526a8a5f05cd660472000012/partners/sendgrid?api_key=081da3fc30ca5b11b1f8b133de7064817b57c86d3261f140dc36243eb6829a100d4c5e38db8dd289194d4e02159bbf7ebad74a1fba1fce7a421a00c7a2f7d0f8ba5ef6ea3cece6345904d79395c35cdeaed221c3a208c897e79dd95da399a1d2d1d3e66768af929aa52d621e5bd296a5"}
+```
+
+
+### Whitelabel the Subuser
```

Comment:
We've recently deprecated the term "whitelabel", can you please use the following:

Whitelabel = Domain Authentication
IP Whitelabel = Reverse DNS
Link Whitelabel = Link branding

Reply:
Is the proper API endpoint to perform these whitelabeling activities still 'https://api.sendgrid.com/v3/whitelabel/ips', 'https://api.sendgrid.com/v3/whitelabel/domains', etc, or do those ```curl``` links need to be changed as well?